src/main/java/com/joejoe2/chat/service/channel/GroupChannelServiceImpl.java

```diff
@@ -116,22 +116,33 @@ else if (subscriber instanceof WebSocketSession)
         });
   }
 
+  private User getUserById(String userId) throws UserDoesNotExist {
```

Comment:
This and the following method occur in both this class and PrivateChannelServiceImpl. Perhaps this could be moved into one place.

Reply:
I have tried that before. But the userRepository interface cannot have this method because default jpa method cannot access data. The only solution is to create another userRepository interface and implement it.

A little confused about worth it or not.

Reply:
These could be moved into another Bean which would be injected into these service classes.

Reply:
Oops, I forgot we have an UserService here.

Reply:
ok, no problem :) it happens